packages/compass-preferences-model/src/storage.ts

```diff
@@ -71,7 +73,19 @@ export class StoragePreferences extends BasePreferencesStorage {
   }
 
   private async readPreferences(): Promise<UserPreferences> {
-    return JSON.parse(await fs.readFile(this.getFilePath(), 'utf8'));
+    const preferences = JSON.parse(
+      await fs.readFile(this.getFilePath(), 'utf8')
+    );
+
+    // Exclude old and renamed preferences so we don't try and save those back.
```

Comment:
I think the problem here is that we're passing off anything that will JSON.parse() without throwing as valid UserPreferences where it could be `[]` or `{}` or partial UserPreferences or a superset or..

I now strip unknown preferences (think preferences we removed or renamed since last time the user's preferences were saved to disk), but we could have also _added_ preferences since then.

(To not even mention edge cases where someone could have edited the file manually.)

But I tried to keep this fix PR small.